### DISCUSSION TITLE
.create() is not working

### DISCUSSION THREAD

**Discussion by BuffKid** [pocketbase/pocketbase#5974] [Q&A]
Hi Everyone,

I'm new to using pocketbase, and I wanted to test out how the database worked. I've followed the documentation, and it lets me do a getList or GET method perfectly. But there is an error in the POST method. When I try running the .create() to create a new record, a POST method is not even sent through the network, and a new record is not created. Can you help me fix this issue, so I can start creating records with my js code.

I am using react and nextjs for my project. I am using js pages not jsx.

**Comment by ganigeorgiev** [comment]
You have better chance someone to assist you if you share a **minimal** code sample illustrating how you are sending the request and elaborate where your code is running in general (client-side, server-side, both?).

Since you've mentioned that your request doesn't reach the PocketBase server, then your issue is most likely caused by either an incorrect JS SDK initialization url or JS error.

The easiest way to rule out PocketBase issue is to create a static `pb_public/index.html` file:

```html
<!DOCTYPE html />
<html>
<head>
    <meta charset="utf-8" />
    <title>Test</title>
</head>
<body>
    <script type="module">
        import PocketBase from "https://cdn.jsdelivr.net/gh/pocketbase/js-sdk@v0.22.0/dist/pocketbase.es.mjs"

        const pb = new PocketBase("http://127.0.0.1:8090");

        // authenticate if necessary
        await pb.collection("users").authWithPassword("test@example.com", "1234567890")

        try {
            await pb.collection("example").create({
                "title": "test"
            })
        } catch (err) {
            // see https://github.com/pocketbase/js-sdk#error-handling
            console.log(err.response, err.originalError)
        }
    </script>
</body>
</html>
```
Adjust the above code to your collection(s), navigate to `http://127.0.0.1:8090` in the browser and check the console.

For more detailed errors you can also start the executable with `--dev` and/or check the _Dashboard > Logs_ section.

**Comment by BuffKid** [comment]
Hi thanks, for trying to help, my code doesn't need any authentication, but the post request for the create command doesn't work. I attached a snippet of my code below:

```
'use client';

import { useState } from 'react';
import { useRouter } from 'next/navigation';
import PocketBase from 'pocketbase'

export default function CreateNote() {
  const [title, setTitle] = useState('');
  const [content, setContent] = useState('');

  const router = useRouter();

  const create = async() => {
    const db = new PocketBase('http://127.0.0.1:8090');
    const data = {title: title, content: content}
    await db.collection('notes').create(data)
 
    setContent('');
    setTitle('');
    router.refresh();
  }

  return (
    <form onSubmit={create}>
      <h3>Create a new Note</h3>
      <input
        type="text"
        placeholder="Title"
        value={title}
        onChange={(e) => setTitle(e.target.value)}
      />
      <textarea
        placeholder="Content"
        value={content}
        onChange={(e) => setContent(e.target.value)}
      />
      <button type="submit">
        Create note
      </button>
    </form>
  );
}
```

**Reply by BuffKid** [reply]
I don't know why, but the code is not showing up as code. it starts from use client to the ending }.

**Reply by ganigeorgiev** [reply]
_Code blocks need to be wrapped in 3 backticks (see [GitHub formatting](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code))._

I cannot reproduce your error and I don't know how to help you further. As last suggestion check the browser console for any JS errors. You can also investigate what happens in the browser's _Inspect tool > Network_ tab when you try to submit your form.

**Reply by BuffKid** [reply]
Thanks, for the tip. I have tried to open the browser's Inspect tool > Network tab, there are only fetch status, there is no post status, even though I'm sending a post status through the create function.

**Reply by ganigeorgiev** [reply]
Based on the formatted code, it looks like your form handler is missing `e.preventDefault()`, aka. try changing it to:

```js
async function create(e) {
    e.preventDefault();

    ...
}
```

**Reply by BuffKid** [reply]
Thanks, i tried that but i still get an error. Here is a image of the error. But there is still no post being sent on the network tab.
![image](https://github.com/user-attachments/assets/8e97334e-677c-4260-8bce-3e6d558da0fd)

**Reply by ganigeorgiev** [reply]
Please check the minimal example at the beginning in https://github.com/pocketbase/pocketbase/discussions/5974#discussioncomment-11414617.

Wrap your code in try catch and console log the error details as shown in the example. I cannot help you further.

**Reply by BuffKid** [reply]
Thanks, I got the error message below if that helps.
![image](https://github.com/user-attachments/assets/7d16a3a1-e238-4df3-ae52-c3552c1f2a24)

**Reply by ganigeorgiev** [reply]
Your issue is not PocketBase related. There is something wrong with your environment but without details and some clear steps to reproduce neither me, nor anyone else can assist you.

`ERR_CONNECTION_REFUSED` is the more important error that you need to investigate why it happens, aka. why your browser cannot connect to the PocketBase server. There could be hundreds of reasons why - from firewall, proxy/vpn misconfiguration, misbehaving browser extension, etc.

I provided a minimal example above that you can test with just PocketBase and you refuse to follow the basic instructions (you also didn't logged the error in your code the way how it is in the example!). Even if I wanted to I just don't know how to help you further and I'm not willing to continue this back-and-forth.

**Reply by BuffKid** [reply]
Thank, for trying to help me out. I did follow the instructions and use that minimal example in my code. I logged the error and that's what i got. I will try to fix the ERR_CONNECTION_REFUSED.
